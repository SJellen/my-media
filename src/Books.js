import React, {useState} from 'react'
import {booksReading, booksRead, handshakeFiveBooks} from './bookList'

function Books() {
      
    const [handshakeReview, setHandshakeReview] = useState()
    const [watchedReview, setWatchedReview] = useState()
    const [watchingReview, setWatchingReview] = useState()
    
    function handleHandshakeReviewClick(e) {
        let selection = handshakeFiveBooks[e].review
        setHandshakeReview(selection)
        if (selection === handshakeReview) {
            setHandshakeReview()
        }
    }

    function handleWatchingReviewClick(e) {
        let selection = booksReading[e].review
        setWatchingReview(selection)
        if (selection === watchingReview) {
            setWatchingReview()
        }
    }

    function handleWatchedReviewClick(e) {
        let selection = booksRead[e].review
        setWatchedReview(selection)
        if (selection === watchedReview) {
            setWatchedReview()
        }
    }

   const bookHandshake = handshakeFiveBooks.map((book, index) => (
        <div className="bookHandshake"  key={book.id}>
            <img src={`http://covers.openlibrary.org/b/isbn/${book.isbn}-M.jpg`} alt="book covers" className="book" />
            <div  className="word_box">
            <h4 className="title">{book.title}</h4>
            <h4 className="star">{Array(book.rating).fill(String.fromCharCode(10029))}</h4>
            { handshakeReview === book.review ? 
                <div>
                <i className="material-icons arrow_drop_up_icon" onClick={() => handleHandshakeReviewClick(index)}
                style={{display: book.review === "" ? "none" : ""}}
                >arrow_drop_up_icon</i>
                <p className="review" style={{display:  book.review ? "block" : "none"}}>{book.review}</p>
                </div>
                 :
                 <i className="material-icons arrow_right_icon" onClick={() => handleHandshakeReviewClick(index)}
                 style={{display: book.review === "" ? "none" : ""}}
                 >arrow_right_icon</i>
            }
            </div>   
        </div>
    ))

    const nowReading = booksReading.map((book, index) => (
        <div className="now_reading"    key={book.id}>
            <img src={`http://covers.openlibrary.org/b/isbn/${book.isbn}-M.jpg`} alt="book covers" className="book" />
            <div  className="word_box">
            <h4 className="title">{book.title}</h4>
            <h4 className="star">{Array(book.rating).fill(String.fromCharCode(10029))}</h4>
             { watchingReview === book.review ? 
                <div>
                <i className="material-icons arrow_drop_up_icon" onClick={() => handleWatchingReviewClick(index)} 
                style={{display: book.review === "" ? "none" : ""}}>arrow_drop_up_icon</i>
                <p className="review" style={{display:  book.review ? "block" : "none"}}>{book.review}</p>
                </div>
                 :
                 <i className="material-icons arrow_right_icon" onClick={() => handleWatchingReviewClick(index)}
                 style={{display: book.review === "" ? "none" : ""}}
                 >arrow_right_icon</i>
            }
            </div>    
        </div>
    ))

    const recentlyRead = booksRead.map((book, index) => (
        <div className="recently_read"  key={book.id}>
            <img src={`http://covers.openlibrary.org/b/isbn/${book.isbn}-M.jpg`} alt="book covers" className="book" />
            <div  className="word_box">
            <h4 className="title">{book.title}</h4>
            <h4 className="star">{Array(book.rating).fill(String.fromCharCode(10029))}</h4>
            { watchedReview === book.review ? 
                <div>
                <i className="material-icons arrow_drop_up_icon" onClick={() => handleWatchedReviewClick(index)} 
                style={{display: book.review === "" ? "none" : ""}}>arrow_drop_up_icon</i>
                <p className="review" style={{display:  book.review ? "block" : "none"}}>{book.review}</p>
                </div>
                 :
                 <i className="material-icons arrow_right_icon" onClick={() => handleWatchedReviewClick(index)}
                 style={{display: book.review === "" ? "none" : ""}}
                 >arrow_right_icon</i>
            }
            </div>   
        </div>
    ))
   
    return (
        <div>
            <h1 className="section_title" id="books">Books</h1>
            <h2 className="sub_section_title">Handshake Five</h2>
            <div className="book_handshake_flex">
            {bookHandshake}
            </div>
            <h2 className="sub_section_title">Currently Reading</h2>
            <div className="currently_reading_flex">
            {nowReading}
            </div>
            <h2 className="sub_section_title">Recently Read</h2>
            <div className="recently_read_flex">
            {recentlyRead}
            </div>
        </div>
    )
}

export default Books