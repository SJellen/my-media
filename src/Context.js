import React, {useState, useEffect} from 'react'


const Context = React.createContext()

function ContextProvider({children}) {

    

    function reviewToggle(e, arr, review, setter) {
        let selection = arr[e].id
        setter(selection)
        if (selection === review) {
            setter()  
        } 
    }

    function filterSet(item, activeReview) {
        const baseFilter = { filter: "grayscale(0%)"}
        const grayscale = { filter: "grayscale(100%)", opacity: .05}
        const hightLight = { filter: "grayscale(0%)"}
        
        if (activeReview === undefined ) {
            return baseFilter
        } else if (item.id === activeReview) {
            return hightLight
        } else if (item.id !== activeReview) {
            return grayscale
        }
    }






    function fullStarMaker(rating) {
        const  ratingSplit= rating.toString().split('.')
        const fullStarNumber = parseInt(ratingSplit[0])
        const full = Array(fullStarNumber).fill(String.fromCharCode(10029)) 
        return full   
    }

    function halfStarMaker(rating) {
        const  ratingSplit= rating.toString().split('.')
        const halfStarNumber = parseInt(ratingSplit[1])
        const half = halfStarNumber === 5 ?  '1/2' : ''
        return half   
    }
    


    return (
        <Context.Provider value={{fullStarMaker, halfStarMaker, reviewToggle,filterSet}}>
            {children}
        </Context.Provider>
    )
}

export {ContextProvider, Context}
