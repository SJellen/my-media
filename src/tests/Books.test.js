import React from 'react'
import { render, fireEvent, cleanup } from '@testing-library/react'
import Books from '../components/Books'
import {ContextProvider} from '../Context'
import Context from '../Context'


afterEach(cleanup)

test('take a snapshot of Books', () => {
    const { asFragment } = render(<ContextProvider value={Context}><Books /></ContextProvider>)
    expect(asFragment(<Books />)).toMatchSnapshot()
})





test('check book component render', () => {
    render(<ContextProvider value={Context}><Books /></ContextProvider>)
})



describe('check each subcomponent renders', () => {
        test('handshakefiveBooks render', () => {
        const { queryByTitle } = render(<ContextProvider value={Context}><Books /></ContextProvider>)
        const handShakeFive = queryByTitle('handShakeFiveBooks')
        expect(handShakeFive).toBeTruthy()
    })
    test('currently reading render', () => {
        const { queryByTitle } = render(<ContextProvider value={Context}><Books /></ContextProvider>)
        const currentlyReading = queryByTitle('currentlyReading')
        expect(currentlyReading).toBeTruthy()
    })
    test('recentlyRead render', () => {
        const { queryByTitle } = render(<ContextProvider value={Context}><Books /></ContextProvider>)
        const recentlyRead = queryByTitle('recentlyRead')
        expect(recentlyRead).toBeTruthy()
    })
})