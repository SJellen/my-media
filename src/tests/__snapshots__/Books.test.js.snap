// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`take a snapshot of Books 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="section_title"
      id="books"
      style="opacity: 1; transition: .5s;"
    >
      Books
    </h1>
    <h2
      class="sub_section_title"
      style="opacity: 1; transition: .5s;"
      title="handShakeFiveBooks"
    >
      Handshake Five
    </h2>
    <div
      class="book_handshake_flex"
    >
      <div
        class="bookHandshake"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/0802130208-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            A Confederacy of Dunces
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭✭✭
            </h4>
            <h4
              class="half-star"
            />
          </span>
          <i
            class="material-icons arrow_right_icon"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
      <div
        class="bookHandshake"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/0061177571-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            Post Office
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭✭
            </h4>
            <h4
              class="half-star"
            >
              1/2
            </h4>
          </span>
          <i
            class="material-icons arrow_right_icon"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
      <div
        class="bookHandshake"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/9780553381344-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            The Bonfire of the Vanities
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭✭
            </h4>
            <h4
              class="half-star"
            >
              1/2
            </h4>
          </span>
          <i
            class="material-icons arrow_right_icon"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
      <div
        class="bookHandshake"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/0345476093-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            The Guns of August
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭✭✭
            </h4>
            <h4
              class="half-star"
            />
          </span>
          <i
            class="material-icons arrow_right_icon"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
      <div
        class="bookHandshake"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/0385333846-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            Slaughterhouse-Five
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭✭
            </h4>
            <h4
              class="half-star"
            >
              1/2
            </h4>
          </span>
          <i
            class="material-icons arrow_right_icon"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
    </div>
    <h2
      class="sub_section_title"
      style="opacity: 1; transition: .5s;"
      title="currentlyReading"
    >
      Currently Reading
    </h2>
    <div
      class="currently_reading_flex"
    >
      <div
        class="now_reading"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/9780300120486-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            Caesar Life of a Colossus
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭✭
            </h4>
            <h4
              class="half-star"
            />
          </span>
          <i
            class="material-icons arrow_right_icon"
            style="display: none;"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
      <div
        class="now_reading"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/9780606107211-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            Heir to the Empire
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭✭
            </h4>
            <h4
              class="half-star"
            />
          </span>
          <i
            class="material-icons arrow_right_icon"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
      <div
        class="now_reading"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/0345332687-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            London Match
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭
            </h4>
            <h4
              class="half-star"
            />
          </span>
          <i
            class="material-icons arrow_right_icon"
            style="display: none;"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
      <div
        class="now_reading"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/9780679746492-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            What It Takes: The Way to the White House
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭✭
            </h4>
            <h4
              class="half-star"
            >
              1/2
            </h4>
          </span>
          <i
            class="material-icons arrow_right_icon"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
      <div
        class="now_reading"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/0441012035-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            Neuromancer
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭
            </h4>
            <h4
              class="half-star"
            >
              1/2
            </h4>
          </span>
          <i
            class="material-icons arrow_right_icon"
            style="display: none;"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
    </div>
    <h2
      class="sub_section_title"
      style="opacity: 1; transition: .5s;"
      title="recentlyRead"
    >
      Recently Read
    </h2>
    <div
      class="recently_read_flex"
    >
      <div
        class="recently_read"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/0393038912-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            Guns, germs, and steel
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭✭
            </h4>
            <h4
              class="half-star"
            />
          </span>
          <i
            class="material-icons arrow_right_icon"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
      <div
        class="recently_read"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/044376256-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            Adventures in the Screen Trade
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭✭
            </h4>
            <h4
              class="half-star"
            />
          </span>
          <i
            class="material-icons arrow_right_icon"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
      <div
        class="recently_read"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/9780060539825-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            Burning Chrome
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭✭
            </h4>
            <h4
              class="half-star"
            />
          </span>
          <i
            class="material-icons arrow_right_icon"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
      <div
        class="recently_read"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/0345314999-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            Mexico set
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭✭
            </h4>
            <h4
              class="half-star"
            />
          </span>
          <i
            class="material-icons arrow_right_icon"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
      <div
        class="recently_read"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/9780345418340-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            Berlin game
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭
            </h4>
            <h4
              class="half-star"
            >
              1/2
            </h4>
          </span>
          <i
            class="material-icons arrow_right_icon"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
      <div
        class="recently_read"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/0446677736-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            The Score
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭✭
            </h4>
            <h4
              class="half-star"
            />
          </span>
          <i
            class="material-icons arrow_right_icon"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
      <div
        class="recently_read"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/9780446677721-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            The Mourner
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭
            </h4>
            <h4
              class="half-star"
            />
          </span>
          <i
            class="material-icons arrow_right_icon"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
      <div
        class="recently_read"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/0140185003-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            The Quiet American
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭✭
            </h4>
            <h4
              class="half-star"
            >
              1/2
            </h4>
          </span>
          <i
            class="material-icons arrow_right_icon"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
      <div
        class="recently_read"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/9780140184938-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            Our Man in Havana
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭✭
            </h4>
            <h4
              class="half-star"
            />
          </span>
          <i
            class="material-icons arrow_right_icon"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
      <div
        class="recently_read"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/9780876852637-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            Factotum
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭
            </h4>
            <h4
              class="half-star"
            />
          </span>
          <i
            class="material-icons arrow_right_icon"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
      <div
        class="recently_read"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/0061177571-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            Post Office
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭✭
            </h4>
            <h4
              class="half-star"
            >
              1/2
            </h4>
          </span>
          <i
            class="material-icons arrow_right_icon"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
      <div
        class="recently_read"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/9780876859261-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            Pulp
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭
            </h4>
            <h4
              class="half-star"
            />
          </span>
          <i
            class="material-icons arrow_right_icon"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
      <div
        class="recently_read"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/1626566747-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            The New Confessions of an Economic Hit Man
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭
            </h4>
            <h4
              class="half-star"
            />
          </span>
          <i
            class="material-icons arrow_right_icon"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
      <div
        class="recently_read"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/0446674672-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            The Outfit
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭✭
            </h4>
            <h4
              class="half-star"
            />
          </span>
          <i
            class="material-icons arrow_right_icon"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
      <div
        class="recently_read"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/0446674664-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            The Man With The Getaway Face
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭✭
            </h4>
            <h4
              class="half-star"
            />
          </span>
          <i
            class="material-icons arrow_right_icon"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
      <div
        class="recently_read"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/0226770990-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            The Hunter
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭✭
            </h4>
            <h4
              class="half-star"
            />
          </span>
          <i
            class="material-icons arrow_right_icon"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
      <div
        class="recently_read"
        style="opacity: 1; transition: .5s;"
      >
        <img
          alt="book covers"
          class="book"
          src="https://covers.openlibrary.org/b/isbn/0345520491-M.jpg"
        />
        <div
          class="word_box"
        >
          <h4
            class="title"
          >
            Dream Team
          </h4>
          <span
            class="star-container"
          >
            <h4
              class="star"
            >
              ✭✭✭
            </h4>
            <h4
              class="half-star"
            >
              1/2
            </h4>
          </span>
          <i
            class="material-icons arrow_right_icon"
          >
            arrow_right_icon
          </i>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
